ghcide/src/Development/IDE/Plugin/Completions.hs

```diff
@@ -140,18 +143,31 @@ getCompletionsLSP ide plId
                 exportsCompls = mempty{anyQualCompls = exportsCompItems}
             let compls = (fst <$> localCompls) <> (fst <$> nonLocalCompls) <> Just exportsCompls <> Just lModules
 
-            pure (opts, fmap (,pm,binds) compls, moduleExports)
+            -- get HieAst if OverloadedRecordDot is enabled
+#if MIN_VERSION_ghc(9,2,0)
+            let uses_overloaded_record_dot (ms_hspp_opts . msrModSummary -> dflags) = xopt LangExt.OverloadedRecordDot dflags
+#else
+            let uses_overloaded_record_dot _ = False
+#endif
+            ms <- fmap fst <$> useWithStaleFast GetModSummaryWithoutTimestamps npath
+            astres <- case ms of
```

Comment:
Possibly stupid question, but why guard here? What happens if we just guard getting the HIE AST on whether or not we have record dot?

Reply:
If I understand you correctly, that's exactly what I'm trying to do here. Maybe there's a simpler way to do this?

Reply:
No, it was a stupid question: I missed that the `uses_overloaded_record_dot` predicate needs the module summary!